src/metabase/api/search.clj

```diff
@@ -106,21 +106,48 @@
 (mu/defn add-collection-join-and-where-clauses
   "Add a `WHERE` clause to the query to only return Collections the Current User has access to; join against Collection
   so we can return its `:name`."
-  [honeysql-query               :- ms/Map
-   collection-id-column         :- keyword?
-   {:keys [current-user-perms]} :- SearchContext]
+  [honeysql-query                                :- ms/Map
+   collection-id-column                          :- keyword?
+   {:keys [current-user-perms
+           filter-items-in-personal-collection]} :- SearchContext]
   (let [visible-collections      (collection/permissions-set->visible-collection-ids current-user-perms)
         collection-filter-clause (collection/visible-collection-ids->honeysql-filter-clause
                                   collection-id-column
                                   visible-collections)
-        honeysql-query           (-> honeysql-query
-                                     (sql.helpers/where collection-filter-clause)
-                                     (sql.helpers/where [:= :collection.namespace nil]))]
+        honeysql-query           (cond-> honeysql-query
+                                   true
+                                   (sql.helpers/where collection-filter-clause [:= :collection.namespace nil])
+
+                                   (some? filter-items-in-personal-collection)
+                                   identity)]
```

Comment:
Isn't this a no-op?

Reply:
yep, I've removed it in https://github.com/metabase/metabase/pull/34887/commits/7544cf02639321ae53d24746304755dcebfe5c86